### ISSUE TITLE
YoloV8 inference post pruning

### ISSUE THREAD

**Issue by getroy74** [nathanhubens/fasterai#14]
Dear Author, While walking through your YoloV8 example, the inference time post pruning is ~28ms while before pruning it is ~8mS. What is the reason behind it?

Rgds
Amit

**Comment by nathanhubens** [comment]
Hi @getroy74 !
Would you have more info on that issue ?

For me, pruning provides a speed-up, here is a quick code for testing:

```
model = YOLO('yolov8l.pt').model
example_inputs = torch.randn(1, 3, 640, 640)
replace_c2f_with_c2f_v2(model)

%%timeit
model(example_inputs)
```


And to test after pruning:
```
for name, param in model.named_parameters():
        param.requires_grad = True
    
        ignored_layers = []
        unwrapped_parameters = []
        for m in model.modules():
            if isinstance(m, (Detect,)):
                ignored_layers.append(m)

pruner = Pruner(model, 0.15, 'local', large_final, ignored_layers=ignored_layers, iterative_steps=1)
pruner.prune_model()
```

This gives me 345ms before and 290ms after pruning. Speed-up may depend on CPU and GPU, as GPU might require a higher batch size to take advantage of compute reduction from pruning.

Hope that helps !
